src/main/java/com/cegeka/academy/service/userChallenge/UserChallengeServiceImpl.java

```diff
@@ -26,4 +31,21 @@
         return userChallengeList.stream().map(userChallenge -> UserChallengeMapper.convertUserChallengeToUserChallengeDTO(userChallenge)).collect(Collectors.toList());
 
     }
+
+    @Override
+    public UserChallenge rateUser(UserChallengeDTO userChallengeDTO, Long ownerId, Long userChallengeId) {
+
+        userChallengeDTO.setId(userChallengeId);
+        UserChallenge userChallenge = userChallengeRepository.findById(userChallengeDTO.getId()).get();
```

Comment:
Aici cred ca puteai folosi direct userChallengeId primit ca parametru. In felul asta nu mai aveai nevoie nici sa faci setId() pe userChallengeDTO.